discovery-infra/test_infra/helper_classes/cluster.py

```diff
@@ -563,53 +574,43 @@ def wait_for_install(
         )
 
     @JunitTestCase()
-    def prepare_for_install(
-            self,
-            nodes,
-            iso_download_path=env_variables['iso_download_path'],
-            iso_image_type=env_variables['iso_image_type'],
-            ssh_key=env_variables['ssh_public_key'],
-            nodes_count=env_variables['num_nodes'],
-            vip_dhcp_allocation=env_variables['vip_dhcp_allocation'],
-            download_image=True,
-            platform=env_variables['platform']
-    ):
-        if download_image:
-            if env_variables.get('static_network_config'):
+    def prepare_for_installation(self, nodes: Nodes, static_network_config=None, **kwargs):
+        self.update_config(**kwargs)
+
+        if self._config.download_image:
+            if static_network_config:
                 static_network_config = static_network.generate_static_network_data_from_tf(nodes.controller.tf_folder)
             else:
                 static_network_config = None
 
             self.generate_and_download_image(
-                iso_download_path=iso_download_path,
-                iso_image_type=iso_image_type,
-                ssh_key=ssh_key,
+                iso_download_path=self._config.iso_download_path,
+                iso_image_type=self._config.iso_image_type,
                 static_network_config=static_network_config
             )
-        nodes.start_all()
-        self.wait_until_hosts_are_discovered(nodes_count=nodes_count, allow_insufficient=True)
-        nodes.set_hostnames(self)
+        nodes.start_all(self._config.static_ips_config)
+        self.wait_until_hosts_are_discovered(allow_insufficient=True)
+        nodes.set_hostnames(self, self._config.nodes_count, self._config.is_ipv6, self._config.static_ips_config)
         if self._high_availability_mode != consts.HighAvailabilityMode.NONE:
-            self.set_host_roles()
+            self.set_host_roles(len(nodes.get_masters()), len(nodes.get_workers()))
         else:
             nodes.set_single_node_ip(self)
         self.set_network_params(
             controller=nodes.controller,
-            vip_dhcp_allocation=vip_dhcp_allocation,
+            vip_dhcp_allocation=self._config.vip_dhcp_allocation,
+            service_network_cidr=self._config.service_network_cidr,
+            cluster_network_cidr=self._config.cluster_network_cidr,
+            cluster_network_host_prefix=self._config.cluster_network_host_prefix
         )
         self.wait_for_ready_to_install()
 
-        if platform == consts.Platforms.NONE:
+        if self._config.platform == consts.Platforms.NONE:
             self._configure_load_balancer(nodes.controller)
 
-    def download_kubeconfig_no_ingress(
-            self, kubeconfig_path=env_variables['kubeconfig_path']
-    ):
+    def download_kubeconfig_no_ingress(self, kubeconfig_path):
```

Comment:
This should have a default value for kubeconfig_path, coming from self._config

Reply:
You can use `cluster.download_kubeconfig_no_ingress(kubeconfig_path=env_variables["kubeconfig_path"])`